src/components/virtualRepeater/virtualRepeater.js

```diff
@@ -0,0 +1,452 @@
+  this.size = this.$attrs.mdHorizontal
+      ? this.$element[0].clientWidth
+      : this.$element[0].clientHeight;
+  this.repeater && this.repeater.containerUpdated();
+};
+
+
+/** @return {number} The container's scrollHeight or scrollWidth. */
+VirtualRepeatContainerController.prototype.getScrollSize = function() {
+  return this.scrollSize;
+};
+
+
+/**
+ * Sets the scrollHeight or scrollWidth. Called by the repeater based on
+ * its item count and item size.
+ * @param {number} The new size.
+ */
+VirtualRepeatContainerController.prototype.setScrollSize = function(size) {
+  if (this.scrollSize !== size) {
+    this.sizer.style[this.$attrs.mdHorizontal ? 'width' : 'height'] = size + 'px';
+  }
+
+  this.scrollSize = size;
+};
+
+
+/** @return {number} The container's current scroll offset. */
+VirtualRepeatContainerController.prototype.getScrollOffset = function() {
+  return this.scrollOffset;
+};
+
+
+VirtualRepeatContainerController.prototype.resetScroll = function() {
+  this.scroller[this.$attrs.mdHorizontal ? 'scrollLeft' : 'scrollTop'] = 0;
+  this.handleScroll_(!!this.$attrs.mdHorizontal);
+};
+
+
+VirtualRepeatContainerController.prototype.handleScroll_ = function(horizontal) {
+  var NUM_EXTRA = 3;
+
+  var transform, offset;
+  if (horizontal) {
+    offset = this.scroller.scrollLeft;
+    transform = 'translateX(';
+  } else {
+    offset = this.scroller.scrollTop;
+    transform = 'translateY(';
+  }
+
+  if (offset === this.scrollOffset) return;
+  this.scrollOffset = offset;
+
+  var itemSize = this.repeater.getItemSize();
+  transform += Math.max(0,
+      Math.floor(this.scrollOffset / itemSize) - NUM_EXTRA) * itemSize + 'px)';
+
+  this.offsetter.style.webkitTransform = transform;
+  this.offsetter.style.transform = transform;
+
+  this.repeater.containerUpdated();
+};
+
+
+/**
+ * @ngdoc directive
+ * @name mdVirtualRepeat
+ * @module material.components.virtualRepeat
+ * @restrict A
+ * @description
+ * `md-virtual-repeat` specifies an element to repeat using virtual scrolling.
+ *
+ * Virtual repeat is a limited substitute for ng-repeat that allows renders only
+ * enough dom nodes to fill the container and recycling them as the user scrolls.
+ *
+ * @usage
+ * <hljs lang="html">
+ * <md-virtual-repeat-container md-horizonal>
+ *   <div md-virtual-repeat="i in items">Hello {{i}}!</div>
+ * </md-virtual-repeat-containr>
+ *
+ * <md-virtual-repeat-container md-horizonal>
+ *   <div md-virtual-repeat="i in items">Hello {{i}}!</div>
+ * </md-virtual-repeat-containr>
+ * </hljs>
+ *
+ * @param {number=} md-size The height or width of the repeated elements (which
+ *     must be identical for each element). TODO: If absent, the directive will attempt to
+ *     measure the computed style of the element at startup.
+ */
+function VirtualRepeatDirective($parse) {
+  return {
+    controller: VirtualRepeatController,
+    priority: 1000,
+    require: ['mdVirtualRepeat', '^mdVirtualRepeatContainer'],
```

Comment:
`^^mdVirtualRepeatContainer` is better (it starts looking at the parent, not the element itself).